keps/sig-cluster-lifecycle/wgs/1815-componentconfig-objectmeta/README.md

```diff
@@ -0,0 +1,649 @@
+
+These may be defined in terms of API maturity, or as something else. The KEP
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the
+functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real world usage
+- N installs
+- More rigorous forms of testing e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least 2 releases between beta and
+GA/stable, since there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a deprecated flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality which deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include [conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+The schema matches a subset of the already GA ObjectMeta schema (for tooling compatibility) and will be implemented at the current version of each component API.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+-->
+
+Existing configurations that lack the ObjectMeta fields will continue to work. As the ComponentConfig API's mature, this will change as sctrict decoders are defaulted without leniancy. 
```

Comment:
We can add a guideline to this proposal that components should always treat ObjectMeta fields as optional.
Strict decoding won't require optional fields to be set, it will just reject unrecognized fields.

1. Not specified, not defined: Common case today.
1. Specified, but not defined in the type: Some users do this today for interop with tools. Enabling strict decoding breaks this. Thus, we must define it in the type to enable strict decoding.
1. Not specified, but defined: ObjectMeta should be kept optional, so that this remains backwards-compatible with "not specified, not defined".
1. Specified, and defined: Works correctly with strict decoding. We need to add ObjectMeta so that strict decoding remains backwards with "specified, but not defined in the type".

Reply:
Added this into the verbiage.